### PR TITLE
fixes #8, enum in object

### LINKED ISSUE #8: (regression) Protobuf.decode cannot works with enum used as objects field's type
it works before, but now it cannot compile

```Nim
import options
import protobuf_serialization

type
  Color {.pure.} = enum
    RED
    GREEN
    
  Apple {.protobuf3.} = object
    color {. fieldNumber: 12.}: Color

var msg: string
let y = Protobuf.decode(msg, Apple)
```

error messages:
```text
Error: Encoding specified for an already wrapped type, or a type which isn't wrappable due to always having one encoding (byte, char, bool, or float).
```

### PR DESCRIPTION
**Pull request by jangko**

### CODE REVIEW

**Comment by jangko** [comment]
Apart from protobuf3 should serialize/deserialize Option[T] or not like @zah commented in #7, this PR focuses on fixing compilation error.

File: protobuf_serialization/writer.nim [hunk 1]
```diff
@@ -173,12 +173,17 @@ proc writeValueInternal[T](stream: OutputStream, value: T) =
               {.fatal: "Encoding pragma specified yet no enoding matched. This should never happen.".}
 
           elif flattenedField is FixedTypes:
-            stream.writeFieldInternal(fieldNum, flattenedField, type(value), fieldName)
-
+            when sizeof(flattenedField) == 4:
+              stream.writeFieldInternal(fieldNum, Float32(flattenedField), type(value), fieldName)
```
Comment (zah) [inline protobuf_serialization/writer.nim]: I'm having difficulties understanding why these `Float` types appear here. We are in the branch for `FixedTypes` which includes integer types as well. Why are we converting them to floats?

Comment (jangko) [reply protobuf_serialization/writer.nim]: sorry, that is my logical error, fixed.
clearly we need anti regression test suite.

**Comment by jangko** [comment]
remove some commits in favor of #11

**Pull request merged**